crates/miden-lib/asm/program_executor/executor.masm

```diff
@@ -0,0 +1,38 @@
+use.std::utils
+
+use.kernel::memory
+use.kernel::prologue
+
+#! 
+proc.main
+    # Prologue
+    # ---------------------------------------------------------------------------------------------
+
+    exec.prologue::prepare_transaction
+    # => []
+
+    # Script Processing
+    # ---------------------------------------------------------------------------------------------
+
+    # get the memory address of the transaction script root and load it to the stack
+    exec.memory::get_tx_script_root_ptr
+    padw dup.4 mem_loadw
+    # => [TX_SCRIPT_ROOT, tx_script_root_ptr]
+
+    exec.utils::is_empty_word not movdn.4 dropw
+    # => [has_tx_script, tx_script_root_ptr]
```

Comment:
Do we need this? I would think that if transaction script is not specified for some reason that this should be an error.